### ISSUE TITLE
Job costing more time than cron interval?

### ISSUE THREAD

**Issue by laixintao** [Tivix/django-cron#140]
What if I have a cron job run every minute, but the job cost 3min to finish?

From the code I think all of the jobs will be delayed, aka, the job actually behave like running every 3min.

Can we add `H:M` to lock name? Like change [this](https://github.com/Tivix/django-cron/blob/7fa5c2d2835520d8c920f3c103937ab292d9e545/django_cron/backends/lock/cache.py#L58) to `f{jobname}-{hour}:{minute}`？ But if so, we have to query the db first, I am not sure if there are any race conditions.

**Comment by laixintao** [comment]
Thanks for reply.

The case is, I have 4 `runcrons` runs parallelly, and I have a cronjob run for every minute, but one job cost 3 minutes.

Let's say, at 00:01, one job started, and, at 00:02, the job hasn't finished, so all other 3 process was blocked. The next run could be at 00:04 ( when the 00:01 job finished).

---

If I turn ALLOW_PArALLEL_RUNS on, then at 00:01, the job will be run 4 times. (because I have 4 `runcron` process)

**Comment by JedrzejMaluszczak** [comment]
You are right " all of the jobs will be delayed, aka, the job actually behave like running every 3min." 
It behaves the same way for `run_at_times`.  If the first task doesn't complete, the second task will be delayed and will run right after the first.
